### PR TITLE
Prohibit simultaneous use of optimize_memory_usage and handle_timeout_termination

### PR DESCRIPTION
**Pull request by MWeltevrede**
Prohibit the simultaneous use of `optimize_memory_usage = True` and `handle_timeout_termination = True` due to a bug discussed in  #934.

## Description
This implements the hotfix as discussed in #934 that will throw an exception if both `optimize_memory_usage = True` and `handle_timeout_termination = True`. 

It also modifies `test_warn_buffer()` that tests whether the model emits a warning if `optimize_memory_usage = True`. The default value of `handle_timeout_termination` is `True` and so this test would encounter the exception implemented as part of this hotfix. To avoid this issue the test is modified to set `handle_timeout_termination = False`, as this does not interfere with the emitting of the warning.  

## Motivation and Context
- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

### CODE REVIEW

File: tests/test_save_load.py [hunk 1]
```diff
@@ -375,6 +375,7 @@ def test_warn_buffer(recwarn, model_class, optimize_memory_usage):
         select_env(model_class),
         buffer_size=100,
         optimize_memory_usage=optimize_memory_usage,
+        replay_buffer_kwargs={"handle_timeout_termination": False},
```
Comment (Miffyli) [inline tests/test_save_load.py:378]: Hmm why this change? If removed, we might not be testing `handle_timeout_termination` anymore. Ideally tests should use the default values.

**Comment by Miffyli** [comment]
LGTM! Just a tiny comment + quick fix for consistent style :)

File: tests/test_save_load.py (hunk 1 above)
Comment (MWeltevrede) [reply tests/test_save_load.py:378]: This change is required because `test_warn_buffer()` runs with `optimize_memory_usage=False` and `optimize_memory_usage=True` (it is designed to test different behaviours depending on the different values of that parameter). In the latter case, the default value of `handle_timeout_termination=True` will cause the `ReplayBuffer` to throw an exception (which is the new behaviour this PR implements). 

I agree it's not ideal we have to change the default values to be able to run this test. However, I think that is inevitable unless we properly fix the bug in #934, or change the default value of `handle_timeout_termination` to `False` (which I think is also not ideal).

**Review by araffin — APPROVED** [review]
